user_notification/models.py

```diff
@@ -24,22 +26,67 @@ class JobAlert(models.Model):
     def __str__(self):
         return str(self.user_account_id)
 
-    # Calculate the date of the next alert
     @classmethod
     def calc_check_date(cls, last_check_date, frequency_in_days):
+        # Calculate the date of the next alert
         if frequency_in_days < 1:
             myError = ValueError(
                 'frequency_in_days should be Greater than or equal to 1')
             raise myError
         else:
-            if isinstance(last_check_date, datetime):
-                res = last_check_date + timedelta(days=frequency_in_days)
-            else:
+            if isinstance(last_check_date, str):
                 last_date = datetime.strptime(
                     last_check_date, "%Y-%m-%d").date()
                 res = last_date + timedelta(days=frequency_in_days)
+            else:
+                res = last_check_date + timedelta(days=frequency_in_days)
             return res
 
+    @classmethod
+    def jobs_date_check(cls, jobs, last_check_date):
+        # Excludes old job posts by comparing their creation date with last check date
+        if isinstance(last_check_date, str):
```

Comment:
can "last_check_date" not be a string?
if it's always a string and you just want to check if it's empty or not, no need for "isinstance()"

Reply:
Even though Python enables it, it's kind of a bad coding practice to change a variable's type at some point in the code /to reuse a var, and can lead to many mistakes later (and it's also confusing to whoever reads your code). 
Just create 2 different variables and then when you pass it, you'll know whether it's string or datetime so you won't need to use "isinstance()"

Reply:
Is it possible then to schedule a short meeting with you to work it out? Since we don't have much time till the demo I want to make sure i fix it as you expect.